parser/src/parser/grammar.lalrpop

```diff
@@ -403,6 +448,8 @@ IntegrityAtom: Expression = {
     <trace_access: TraceBindingAccessWithOffset> => Expression::TraceBindingAccess(trace_access),
     <list_folding_type: ListFoldingType<IntegrityExpr>> =>
         Expression::ListFolding(list_folding_type),
+    "$" <function_name: Identifier> "(" <args: CommaElems<IntegrityVariableType>> ")" =>
+        Expression::FunctionCall(function_name, args),
```

Comment:
I don't think we want to use `IntegrityVariableType` for the args here. If we were to adjust the nomenclature of everything to some form of `Binding` and `BindingAccess` (like we've done for `TraceBinding` and `TraceBindingAccess`), then these args would be an array of `BindingAccess`. That means we could use `a[1]` as an argument. However, this parsing rule expects something like `[a]` instead, which I don't think matches what we want.

Hopefully fixing this will help resolve the other issue so that the `$` can be removed as well.

Also, I am finding the name `VariableType` increasingly confusing, especially in this PR. Maybe we should rename this to `VariableValue`. What do you think?